contracts/modules/cover/Cover.sol

```diff
@@ -301,7 +316,7 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     uint coverId,
     BuyCoverParams memory buyCoverParams,
     PoolAllocationRequest[] memory poolAllocations
-  ) external payable onlyMember whenNotPaused {
+  ) external payable onlyMember whenNotPaused nonReentrant {
```

Comment:
remove the require below:

`require(lastCoverSegment.start + lastCoverSegment.period > block.timestamp, "Cover: cover expired");`

Reply:
removed.

Changed condition in the IF to check for that.